napari/utils/theme.py

```diff
@@ -29,38 +46,65 @@ def __getattr__(attr):
     raise AttributeError
 
 
-_themes = {
-    'dark': {
-        'folder': 'dark',
-        'background': 'rgb(38, 41, 48)',
-        'foreground': 'rgb(65, 72, 81)',
-        'primary': 'rgb(90, 98, 108)',
-        'secondary': 'rgb(134, 142, 147)',
-        'highlight': 'rgb(106, 115, 128)',
-        'text': 'rgb(240, 241, 242)',
-        'icon': 'rgb(209, 210, 212)',
-        'warning': 'rgb(153, 18, 31)',
-        'current': 'rgb(0, 122, 204)',
-        'syntax_style': 'native',
-        'console': 'rgb(0, 0, 0)',
-        'canvas': 'black',
-    },
-    'light': {
-        'folder': 'light',
-        'background': 'rgb(239, 235, 233)',
-        'foreground': 'rgb(214, 208, 206)',
-        'primary': 'rgb(188, 184, 181)',
-        'secondary': 'rgb(150, 146, 144)',
-        'highlight': 'rgb(163, 158, 156)',
-        'text': 'rgb(59, 58, 57)',
-        'icon': 'rgb(107, 105, 103)',
-        'warning': 'rgb(255, 18, 31)',
-        'current': 'rgb(253, 240, 148)',
-        'syntax_style': 'default',
-        'console': 'rgb(255, 255, 255)',
-        'canvas': 'white',
+class ThemesDict(EventedDict):
```

Comment:
The `EventedDict` is fantastic, and it'd be great if you could break it out into it's own PR so we can fill out our evented containers.

I'm certain that class will be used.  As mentioned in the meeting though, I wonder if we might be better served by making our `Theme` objects subclass from pydantic.  It even has a lovely `Color` field that we can use:

```python
from napari.utils.events import EventedModel
from pydantic.color import Color


class Theme(EventedModel):
    name: str
    syntax_style: str = 'native'
    background: Color = 'rgb(38, 41, 48)'
    foreground: Color = 'rgb(65, 72, 81)'
    primary: Color = 'rgb(90, 98, 108)'
    secondary: Color = 'rgb(134, 142, 147)'
    highlight: Color = 'rgb(106, 115, 128)'
    text: Color = 'rgb(240, 241, 242)'
    icon: Color = 'rgb(209, 210, 212)'
    warning: Color = 'rgb(153, 18, 31)'
    current: Color = 'rgb(0, 122, 204)'
    console: Color = 'rgb(0, 0, 0)'
    canvas: Color = 'black'
```

```py
In [27]: t = Theme(name='whatever')

In [28]: t.events.foreground.connect(lambda e: print(e.value))

In [29]: t.foreground = 'azure'
azure

In [30]: t.foreground
Out[30]: Color('azure', rgb=(240, 255, 255))

In [31]: t.foreground.as_rgb()
Out[31]: 'rgb(240, 255, 255)'
```

what do you think?

Reply:
Thanks! I will make a separate PR for the `EventedDict` and update the underlying theme model. I was planning on using custom color converter (`transform_color` -> `rgb(X, Y, Z)`) but I think the pydantic's `Color` will work nicely! 😄

Reply:
yeah, I'd love to see us use pydantic's `Color` field more liberally throughout the codebase.  In many cases, it could pretty much replace our [`standardize_color`](https://github.com/napari/napari/blob/master/napari/utils/colormaps/standardize_color.py) module ... and I suspect it's more tested by the community

Reply:
> yeah, I'd love to see us use pydantic's Color field more liberally throughout the codebase. In many cases, it could pretty much replace our standardize_color module ... and I suspect it's more tested by the community

Yes - absolutely agree here. Maybe this is something @andy-sweet can look into too as he looks at some more of the color handling. It could remove a lot of our custom code